src/DssDirectDepositHub.t.sol

```diff
@@ -0,0 +1,863 @@
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.6.12;
+
+import "ds-test/test.sol";
+import "./tests/interfaces/interfaces.sol";
+
+import {DssDirectDepositHub, D3MPoolLike, D3MPlanLike} from "./DssDirectDepositHub.sol";
+
+import {D3MTestPool} from "./tests/stubs/D3MTestPool.sol";
+import {D3MTestPlan} from "./tests/stubs/D3MTestPlan.sol";
+import {D3MTestGem} from "./tests/stubs/D3MTestGem.sol";
+import {D3MTestRewards} from "./tests/stubs/D3MTestRewards.sol";
+import {ValueStub} from "./tests/stubs/ValueStub.sol";
+
+interface Hevm {
+    function warp(uint256) external;
+
+    function store(
+        address,
+        bytes32,
+        bytes32
+    ) external;
+
+    function load(address, bytes32) external view returns (bytes32);
+}
+
+contract DssDirectDepositHubTest is DSTest {
+    uint256 constant WAD = 10**18;
+    uint256 constant RAY = 10**27;
+    uint256 constant RAD = 10**45;
+
+    Hevm hevm;
+
+    VatLike vat;
+    EndLike end;
+    D3MTestRewards rewardsClaimer;
+    DaiLike dai;
+    DaiJoinLike daiJoin;
+    D3MTestGem testGem;
+    TokenLike testReward;
+    SpotLike spot;
+    TokenLike weth;
+    address vow;
+    address pauseProxy;
+
+    bytes32 constant ilk = "DD-DAI-TEST";
+    DssDirectDepositHub directDepositHub;
+    D3MTestPool d3mTestPool;
+    D3MTestPlan d3mTestPlan;
+    ValueStub pip;
+
+    function setUp() public {
+        hevm = Hevm(
+            address(bytes20(uint160(uint256(keccak256("hevm cheat code")))))
+        );
+
+        vat = VatLike(0x35D1b3F3D7966A1DFe207aa4514C12a259A0492B);
+        end = EndLike(0xBB856d1742fD182a90239D7AE85706C2FE4e5922);
+        dai = DaiLike(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+        daiJoin = DaiJoinLike(0x9759A6Ac90977b93B58547b4A71c78317f391A28);
+        spot = SpotLike(0x65C79fcB50Ca1594B025960e539eD7A9a6D434A3);
+        weth = TokenLike(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+        vow = 0xA950524441892A31ebddF91d3cEEFa04Bf454466;
+        pauseProxy = 0xBE8E3e3618f7474F8cB1d074A26afFef007E98FB;
+
+        // Force give admin access to these contracts via hevm magic
+        _giveAuthAccess(address(vat), address(this));
+        _giveAuthAccess(address(end), address(this));
+        _giveAuthAccess(address(spot), address(this));
+
+        testGem = new D3MTestGem(18);
```

Comment:
add test to wind/unwind with non 18 decimals

Reply:
@godsflaw @brianmcmichael @naszam @talbaneth and @gbalabasquer added this test here 74d1971 . Let me know if that covers the cases we were worried about